### PR TITLE
Update clippy and miri submodule

### PR DESCRIPTION
**Pull request by oli-obk**
r? @Manishearth 

cc @kennytm I needed to touch the miri submodule's Cargo.toml to make sure that clippy gets the newest compiletest_rs. This will not fix miri, but since I touched the miri submodule, will this PR fail?

miri is unfixable until #46882 is merged

### CODE REVIEW

**Comment by kennytm** [comment]
Oops. Yes this will fail 🤔

**Comment by Manishearth** [comment]
r=me when it's fixed

File: src/bootstrap/tool.rs [hunk 1]
```diff
@@ -469,7 +469,8 @@ tool_extended!((self, builder),
             target: builder.build.build,
         });
     };
-    Miri, miri, "src/tools/miri", "miri", {};
+    // FIXME(oli-obk) reenable after #46882
+    //Miri, miri, "src/tools/miri", "miri", {};
```
Comment (kennytm) [inline src/bootstrap/tool.rs]: This line shouldn't need to be commented out.

**Comment by oli-obk** [comment]
travis likes it now

**Comment by oli-obk** [comment]
Hmmm... clippy's appveyor works, rustcs fails with 

```
error: linking with `link.exe` failed: exit code: 1561
  |
  = note: "C:\\Program Files (x86)\\Microsoft Visual Studio 14.0\\VC\\bin\\amd64\\link.exe" "/NOLOGO" "/NXCOMPAT" "/LIBPATH:C:\\projects\\rust\\build\\x86_64-pc-windows-msvc\\stage2\\lib\\rustlib\\x86_64-pc-windows-msvc\\lib" "/tmp/clippy_test_build_base\\enum-glob-import-crate.enum_glob_import_crate0.rcgu.o" "/tmp/clippy_test_build_base\\enum-glob-import-crate.enum_glob_import_crate1.rcgu.o" "/tmp/clippy_test_build_base\\enum-glob-import-crate.enum_glob_import_crate2.rcgu.o" "/OUT:/tmp/clippy_test_build_base\\enum-glob-import-crate.stage-id.exe" "/tmp/clippy_test_build_base\\enum-glob-import-crate.crate.allocator.rcgu.o" "/OPT:REF,NOICF" "/DEBUG" "/LIBPATH:/tmp/clippy_test_build_base" "/LIBPATH:C:\\projects\\rust\\build\\x86_64-pc-windows-msvc\\stage2-tools\\release" "/LIBPATH:C:\\projects\\rust\\build\\x86_64-pc-windows-msvc\\stage2-tools\\release/deps" "/LIBPATH:/tmp/clippy_test_build_base\\enum-glob-import-crate.stage-id.aux" "/LIBPATH:C:\\projects\\rust\\build\\x86_64-pc-windows-msvc\\stage2\\lib\\rustlib\\x86_64-pc-windows-msvc\\lib" "/LIBPATH:C:\\projects\\rust\\build\\x86_64-pc-windows-msvc\\stage2\\lib\\rustlib\\x86_64-pc-windows-msvc\\lib" "std-8c525ca5930ac7cd.dll.lib" "C:\\projects\\rust\\build\\x86_64-pc-windows-msvc\\stage2\\lib\\rustlib\\x86_64-pc-windows-msvc\\lib\\libcompiler_builtins-3d7fcf367f7cd6b3.rlib" "advapi32.lib" "ws2_32.lib" "userenv.lib" "shell32.lib" "msvcrt.lib"
  = note: LINK : warning LNK4044: unrecognized option '/tmp/clippy_test_build_base\enum-glob-import-crate.enum_glob_import_crate0.rcgu.o'; ignored
          LINK : warning LNK4044: unrecognized option '/tmp/clippy_test_build_base\enum-glob-import-crate.enum_glob_import_crate1.rcgu.o'; ignored
          LINK : warning LNK4044: unrecognized option '/tmp/clippy_test_build_base\enum-glob-import-crate.enum_glob_import_crate2.rcgu.o'; ignored
          LINK : warning LNK4044: unrecognized option '/tmp/clippy_test_build_base\enum-glob-import-crate.crate.allocator.rcgu.o'; ignored
          LINK : warning LNK4001: no object files specified; libraries used
          LINK : warning LNK4068: /MACHINE not specified; defaulting to X64
          LINK : fatal error LNK1561: entry point must be defined
          
```

I guess there's no `/tmp` directory on windows.

**Comment by oli-obk** [comment]
I am dumping the files in `env!("OUT_DIR")` now, so this should work on any platform

**Comment by kennytm** [comment]
Legit.

```
[01:18:51] error[E0063]: missing field `approximate` in initializer of `rustc_errors::CodeSuggestion`
[01:18:51]    --> tools/clippy/clippy_lints/src/utils/mod.rs:636:16
[01:18:51]     |
[01:18:51] 636 |     let sugg = rustc_errors::CodeSuggestion {
[01:18:51]     |                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^ missing `approximate`
[01:18:51] 
[01:18:51] error[E0063]: missing field `approximate` in initializer of `rustc_errors::CodeSuggestion`
[01:18:51]    --> tools/clippy/clippy_lints/src/utils/mod.rs:636:16
[01:18:51]     |
[01:18:51] 636 |     let sugg = rustc_errors::CodeSuggestion {
[01:18:51]     |                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^ missing `approximate`
[01:18:51] 
[01:18:55] error: aborting due to previous error
[01:18:55] 
[01:18:55] error: Could not compile `clippy_lints`.
```

**Comment by pietroalbini** [comment]
@oli-obk ping from triage!

**Comment by oli-obk** [comment]
We are green 🎉 https://rust-lang-nursery.github.io/rust-toolstate/

**Pull request merged**